omnigibson/action_primitives/starter_semantic_action_primitives.py

```diff
@@ -1642,11 +1646,38 @@ def _navigate_to_pose_direct(self, pose_2d, low_precision=False):
             else:
                 action = self._empty_action()
                 if self._base_controller_is_joint:
-                    direction_vec = body_target_pose[0][:2] / np.linalg.norm(body_target_pose[0][:2]) * m.KP_LIN_VEL
-                    base_action = [direction_vec[0], direction_vec[1], 0.0]
-                    action[self.robot.controller_action_idx["base"]] = base_action
+                    base_action_size = self.robot.controller_action_idx["base"].size
+                    # Joint controller
+                    if base_action_size == 3:
+                        # [x, y, theta]
+                        direction_vec = (
+                            body_target_pose[0][:2]
+                            / np.linalg.norm(body_target_pose[0][:2])
+                            * m.KP_LIN_VEL[type(self.robot)]
+                        )
+                        base_action = [direction_vec[0], direction_vec[1], 0.0]
+                        action[self.robot.controller_action_idx["base"]] = base_action
+                    elif base_action_size == 4:
+                        # [wheel0, wheel1, wheel2, wheel3]
+                        direction_vec = body_target_pose[0][:2]
+                        distance = np.linalg.norm(direction_vec)
+                        # Move forward
+                        linear_x = min(distance, m.KP_LIN_VEL[type(self.robot)])
+                        wheel_velocities = self._calculate_wheel_velocities(linear_x, 0.0, 4)
+                        action[self.robot.controller_action_idx["base"]] = wheel_velocities
+                    elif base_action_size == 2:
+                        # [left, right]
+                        direction_vec = body_target_pose[0][:2]
+                        distance = np.linalg.norm(direction_vec)
+                        # Move forward
+                        linear_x = min(distance, m.KP_LIN_VEL[type(self.robot)])
+                        wheel_velocities = self._calculate_wheel_velocities(linear_x, 0.0, 2)
+                        action[self.robot.controller_action_idx["base"]] = wheel_velocities
+                    else:
+                        raise ValueError("Invalid base action size")
```

Comment:
for these three different actions sizes (2, 3, 4), do they necessarily correspond to the comments? e.g. `[x,y,theta]`? Seems like a strong assumption.